frontend/review/src/components/GraphBox.js

```diff
@@ -0,0 +1,501 @@
+  padding: 35px;
+  background-color: #ffffff;
+  box-shadow: 0 0 14px 0 rgba(0, 0, 0, 0.07);
+  margin-bottom: 30px;
+  min-height: 500px;
+  flex: 1;
+`;
+
+const genAverageData = (seriesData) => {
+  let averageData = [];
+  let windowSize = 0.05;
+  for (let i = 0; i < seriesData.length; i++) {
+    let xVal = seriesData[i][0];
+    let total = 0;
+    let numInTotal = 0;
+    let j = i;
+    while (j >= 0 && seriesData[j][0] >= xVal - windowSize) {
+      total += seriesData[j][1];
+      numInTotal += 1;
+      j -= 1;
+    }
+    let movingAverageVal = total / numInTotal;
+    averageData.push({
+      x: (xVal * 100).toFixed(2),
+      y: movingAverageVal.toFixed(2),
+    });
+  }
+
+  return averageData;
+};
+
+//PCA Demand Chart Data
+const genDemandChartData = (data, averageData) => {
+  return {
+    datasets: [
+      {
+        type: "line",
+        label: "5% Moving Average",
+        data: averageData,
+        borderColor: "#6378E9",
+        borderWidth: 3,
+        fill: false,
+        linear: true,
+      },
+      {
+        type: "line",
+        label: "Registration Difficulty",
+        data: data.map((point) => {
+          return {
+            x: (point[0] * 100).toFixed(2),
+            y: Math.round(point[1] * 100) / 100,
+          };
+        }),
+        backgroundColor: "#AAACEC",
+        borderWidth: 0,
+        steppedLine: true,
+      },
+    ],
+  };
+};
+
+//Percentage of Sections Open Chart Data
+const genPercentChartData = (data) => {
+  return {
+    datasets: [
+      {
+        type: "line",
+        label: "% of Sections Open",
+        data: data.map((point) => {
+          return {
+            x: Math.round((point[0] * 100).toFixed()),
+            y: Math.round(point[1] * 100),
+          };
+        }),
+        borderColor: "#87BD99",
+        backgroundColor: "#E0EBEC",
+        borderWidth: 3,
+        fill: true,
+        steppedLine: true,
+      },
+    ],
+  };
+};
+
+const demandChartOptions = {
+  tooltips: {
+    mode: "index",
+    intersect: false,
+    backgroundColor: "#E9E9E9",
+    bodyFontColor: "#000000",
+    titleFontColor: "#000000",
+    bodyFontSize: 12,
+    cornerRadius: 3,
+    bodySpacing: 3,
+    callbacks: {
+      title: (toolTipItem, data) => {
+        return (
+          data["datasets"][0]["data"][toolTipItem[0]["index"]].x +
+          "% Through Add/Drop"
+        );
+      },
+      beforeBody: (toolTipItem, data) => {
+        let bodyText = "";
+        bodyText +=
+          "Projected Date: " +
+          calcApproxDate(
+            addDropDate.start,
+            addDropDate.end,
+            data["datasets"][0]["data"][toolTipItem[0]["index"]].x / 100
+          ) +
+          "\n";
+        bodyText +=
+          "Registration Difficulty: " +
+          data["datasets"][0]["data"][toolTipItem[0]["index"]].y +
+          "\n";
+        bodyText +=
+          "5% Moving Average: " +
+          data["datasets"][1]["data"][toolTipItem[0]["index"]].y;
```

Comment:
would it be possible to use a js format string to generate this instead of string concats?